### PR TITLE
fix feedstock tests

### PR DESCRIPTION
**Pull request by cdeepali**
## Checklist before submitting

- [x] Did you read the [contributor guide](https://github.com/open-ce/open-ce/blob/main/CONTRIBUTING.md)?
- [ ] Did you update any affected [documentation](https://github.com/open-ce/open-ce/blob/main/doc/)?
- [ ] Did you write any tests to validate this change?

## Description

Fixes # (issue).

## Review process to land 

1. All tests and other checks must succeed.
2. At least one [maintainer](https://github.com/open-ce/open-ce/blob/main/MAINTAINERS.md) must review and approve.
3. If any  [maintainer](https://github.com/open-ce/open-ce/blob/main/MAINTAINERS.md) requests changes, they must be addressed.

### CODE REVIEW

File: lightning/0301-update-codecov-version.patch [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ pytest-cov==4.0.0
+ pytest-forked==1.4.0
+@@ -8,8 +8,8 @@ pytest-rerunfailures==10.3
+ # needed in tests
+ cloudpickle>=1.3, <2.3.0
+ scikit-learn>0.22.1, <1.2.1
+-onnx<1.14.0
+-onnxruntime<1.14.0
++#onnx<1.14.0
++#onnxruntime<1.14.0
```
Comment (alexlang74) [inline lightning/0301-update-codecov-version.patch:28]: Why did you comment that, and not remove directly?

**Review by alexlang74 — APPROVED** [review]

File: lightning/0301-update-codecov-version.patch (hunk 1 above)
Comment (cdeepali) [reply lightning/0301-update-codecov-version.patch:28]: Correct, can be removed. Will remove it.

File: tests/open-ce-tests.yaml [hunk 2]
```diff
@@ -8,7 +8,8 @@ tests:
         git clone https://github.com/PyTorchLightning/pytorch-lightning.git
         cd pytorch-lightning
         git checkout ${PL_COMMIT}
-        conda install -y "scikit-learn>=0.22.2, <1.1.3" regex typed-ast pandas=1.4 scikit-image coverage cryptography psutil
+        git apply ${FEEDSTOCK_DIR}/lightning/0301-update-codecov-version.patch
+        conda install -y scikit-learn regex typed-ast pandas=1.4 scikit-image coverage cryptography psutil
```
[the remark this answers is not in the document]
Comment (cdeepali) [reply tests/open-ce-tests.yaml]: Added version explicitly here, although its included in generated env too.

File: lightning/0301-update-codecov-version.patch [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+-codecov==2.1.12
++codecov==2.1.13
+ pytest==7.2.0
+ pytest-cov==4.0.0
+ pytest-forked==1.4.0
+@@ -8,8 +8,6 @@ pytest-rerunfailures==10.3
+ # needed in tests
+ cloudpickle>=1.3, <2.3.0
+ scikit-learn>0.22.1, <1.2.1
+-onnx<1.14.0
```
Comment (npanpaliya) [inline lightning/0301-update-codecov-version.patch:25]: Why these are removed? Our versions of onnx and onnxruntime fit into this requirement constraint.

Comment (cdeepali) [reply lightning/0301-update-codecov-version.patch:25]: there was a failure when we have these being installed for tests. So I have removed these as we are not running tests that need them.

Comment (npanpaliya) [reply lightning/0301-update-codecov-version.patch:25]: okay

**Review by npanpaliya — APPROVED** [review]

**Comment by cdeepali** [comment]
Merging the PR as it has only test specific changes so build number is not updated.

**Pull request merged**